execution_engine/src/core/engine_state/genesis.rs

```diff
@@ -224,9 +224,11 @@ pub struct ExecConfig {
     accounts: Vec<GenesisAccount>,
     wasm_config: WasmConfig,
     validator_slots: u32,
+    auction_delay: u64,
```

Comment:
Does this really need to be `u64`?

Reply:
The constant was defined as `u64` so I retained it

Reply:
Seems like u32 should be (more than) sufficient to me.

Reply:
I'd like to retain it as `u64` as `auction_delay` is passed in as `u64` to multiple different parts

Reply:
It seems questionable to me that we are using a u64 to track a value that is currently "3" and would never conceivably be particularly large afaik. @mpapierski can you offer any insight?

Reply:
@EdHastingsCasperLabs You're probably right that `AUCTION_DELAY` range is a bit larger than needed. Choice of u64 here is a side effect of also having `EraId` defined as `u64`. It was convenience if anything to avoid casting AUCTION_DELAY to `EraId`(aka u64) when doing arithmetic on era ids. 

I think we can change both EraId to be u32 and also AUCTION_DELAY to be u32. As according to Mateusz on slack with 30min eras that would be ~245146 years until we will saturate the counter and after that time we can change it to u64

Reply:
Changing these values, from `u64` to `u32` I'd argue is out of scope for this PR. @EdHastingsCasperLabs , @mpapierski

Reply:
A small follow up PR/ or ticketing it is fine for me @darthsiroftardis

Reply:
Cool, I shall mark this as resolved then